packages/transaction-controller/src/TransactionController.ts

```diff
@@ -739,16 +865,48 @@ export class TransactionController extends BaseControllerV1<
     };
   }
 
-  startIncomingTransactionPolling() {
-    this.incomingTransactionHelper.start();
+  startIncomingTransactionPolling(networkClientIds: NetworkClientId[] = []) {
+    if (networkClientIds.length === 0 || !this.enableMultichain) {
+      this.incomingTransactionHelper.start();
```

Comment:
This is a great point. We probably could have gone this route from the start, but our goal was to get this in as an addition rather than a whole controller rewrite in order to be as least invasive as possible. In the end we will be stripping out the globally selected network which will allow us to get rid of all these guards and fallback logic. I guess I hope it's okay for this short lived tech debt to exist for now(as it does with the many other controllers we have already modified up to this point).

Reply:
Hey Elliot, yes we considered this approach you're pointing to here. While the direction we've taken does introduce a bit more complexity we generally tried to avoid disrupting the existing non-multichain flows as much as possible. That meant leaving the class variable instantiations of the helpers which use the proxied provider and blockTracker rather than those from the registry. This seemed like the safest intermediate step given that the registry has not been used in production yet and using it in the way you suggest would introduce it even with the multichain feature flag off. I suppose we could instantiate these helpers in the tracking map with the proxied provider/blockTracker but that feels a bit strange as well. Overall we think this approach is least risky and disruptive to existing flows.